src/jsonapi-datastore.js

```diff
@@ -100,6 +101,17 @@ class JsonApiDataStore {
    * @param {object} model The model to destroy.
    */
   destroy(model) {
+    this.graph[model._type][model.id]._associations.map(function(assoc) {
+      if (self.graph[assoc.type][assoc.id][assoc.relation].constructor === Array) {
+        self.graph[assoc.type][assoc.id][assoc.relation].forEach(function(val, idx) {
+          if (val.id === model.id) {
+            self.graph[assoc.type][assoc.id][assoc.relation].splice(idx, 1);
+          }
+        });
+      } else if (self.graph[assoc.type][assoc.id][assoc.relation].id === model.id) {
+        self.graph[assoc.type][assoc.id][assoc.relation] = new JsonApiDataStoreModel(self.graph[assoc.type][assoc.id][assoc.relation]._type);
```

Comment:
I think it would be more consistent with [what we do there](https://github.com/beauby/jsonapi-datastore/pull/28/files#diff-5e059c4f1ce067f813c655ac3a8775c0R183).